doc/manuals/devel/HttpHeaders.md

```diff
@@ -0,0 +1,354 @@
+13) Access-Control-Allow-Methods
+14) Access-Control-Max-Age                                 
+15) Access-Control-Expose-Headers
+16) Allow                                                                                    
+17) Location                                                 
+18) Ngsiv2-AttrsFormat                                    
+19) Fiware-Service
+20) Fiware-Servicepath
+21) Fiware-Total-Count
+22) Fiware-Correlator                                     
+
+23) X-Auth-Token
+
+
+---------------------------------------------------------------------------------------------------------------------
+
+# HTTP Headers used by Orion with their Description:
+
+--------------------------------------------------------------------------------------------------------------------
+
+
+## 1. Accept  : 
+
+Accept HTTP Header is used in Request URL. 
+Accept HTTP Header is used with the request URL for accepting the response in well defined format. 
+For example	in request URL if we type:
+	
+		(e.g):      curl ... 'Accept: applicattion/json'
+		 
+On this request we'll get the Response data in well defined Json format.
+
+The API response payloads in this specification are based on `application/json` and  (for attribute value 
+type operation) `text/plain` MIME types. Clients issuing HTTP requests with accept types different 
+than those will get a `406 Not Acceptable` error.
+
+
+
+## 2. Content-Length :
+
+This header is used by both request and response.
+Content-length HTTP header is a mandatory header for the response of the orion.
+it defines the length of the request and response body in octets (8-bit bytes).
+Orion Context Broker expects always a Content-Length header in all client requests, otherwise the client will receive a "411 Length Required" response. 
+This is due to the way the underlying HTTP library (microhttpd) works.
+	
+		(e.g.): 	 Content-Length: 34	(in response)
+
+
+ 
+## 3. Content-Type : 
+
+It is used to send the data or request in well defined format that can be Json or XML etc.
+syntax is:
+
+		(e.g):      curl ... 'Content-Type: application/json'
+
+this request will send data in well defined JSON format.
+if the Content-Type header is application/xml i.e. ('Content-Type: application/xml'), then the payload must correspond to a well-formed XML document. 
+Specifically, if the resulting URL after applying the template is malformed, then no notification is sent.
+
+
+ 
+ ## 4. Connection :
+ 
+ This Header is used in both Request and Response of Orion.
+ It is used as Control options for the current connection and list of hop-by-hop response fields.
+	
+		(e.g):	    Connection: Keep-Alive
+    
+ When we run request URL with the above Header it will keep connection alive with the host/server.	
+	
+	
+  
+## 5. Expect :
+
+Expect HTTP Header is used as Request Header in Orion.
+Test cases admits two types of 'expect sections', we have to use one. Either
+			--EXPECT--                     
+	   (or) --REG-EXPECT--
+	
```

Comment:
I don't understand what you want to tell here... Under my understanding there is no relation between Expect HTTP header and EXPECT and REG-EXPECT constructs, that are related with the testing framework.

Could you elaborate a bit on this a little bit, please?